From 917e943df2d10d9f8fcc04a5a627f51476c10eca Mon Sep 17 00:00:00 2001
From: Timple Raj M <quic_timple@quicinc.com>
Date: Tue, 30 Jan 2024 16:57:36 +0530
Subject: [PATCH] drivers: mtd: spi-nor: Add MX25U12832F SPI NOR device support

This patch is snapshot of the commit
4104cbaaf87892cec65c2b9fbc2ae59fa820a17b
from win.linuxopenwrt.2.1 branch and
oss/kernel/linux-ipq-5.4 project

Change-Id: Idcba9cd3f56186e96044b9fa55e1bad0c4d22128
Signed-off-by: Timple Raj M <quic_timple@quicinc.com>
---
 drivers/mtd/spi-nor/macronix.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/mtd/spi-nor/macronix.c b/drivers/mtd/spi-nor/macronix.c
index 85eedb3db9ea..9035281eb9e6 100644
--- a/drivers/mtd/spi-nor/macronix.c
+++ b/drivers/mtd/spi-nor/macronix.c
@@ -71,9 +71,15 @@ static const struct flash_info macronix_nor_parts[] = {
 	{ "mx25r3235f",  INFO(0xc22816, 0, 64 * 1024,  64)
 		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
 			      SPI_NOR_QUAD_READ) },
+#ifdef CONFIG_IPQ_FLASH_16M_PROFILE
+	{ "mx25u12832f", INFO(0xc22538, 0, 4 * 1024, 4096)
+		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
+			      SPI_NOR_QUAD_READ) },
+#else
 	{ "mx25u12835f", INFO(0xc22538, 0, 64 * 1024, 256)
 		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
 			      SPI_NOR_QUAD_READ) },
+#endif
 	{ "mx25l25635e", INFO(0xc22019, 0, 64 * 1024, 512)
 		NO_SFDP_FLAGS(SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ)
 		.fixups = &mx25l25635_fixups },
-- 
2.34.1

